### ISSUE TITLE
Requires not working?

### ISSUE THREAD

**Issue by TheOddler** [go-task/task#1264]
- Task version: 3.24.0
- Operating system: NixOS (23.05)
- Experiments enabled: None

I was trying out Task with a simple taskfile:

```
version: "3"

tasks:
  default:
    cmds:
      - task --list-all

  hello:
    cmds:
      - echo 'Hello World from Task! {{.NAME}} - {{.MATCH}}'
    requires:
      vars: [NAME, MATCH]
```

However, when I run `task hello` it just works, rather than give me an error I didn't set the `NAME` nor `MATCH` variable:

```
❯ task hello
task: [hello] echo 'Hello World from Task!  - '
Hello World from Task!  - 
```

**Comment by andreynering** [comment]
Hi @TheOddler,

You version is too old. `requires` is available since v1.27.0. You need to upgrade.

**Comment by TheOddler** [comment]
I'm using the version on nixpkgs right now, supposedly that's 3.24.0, but when I run `task --version` it says:

```
❯ task --version
Task version: (devel)
```

So not sure what that's about :thinking:

**Comment by pd93** [comment]
@TheOddler @andreynering I also noticed this recently as I also have a project with Nix that has Task pinned to 3.24. Looks like the version wasn't getting set properly for a 3.24/25 when we built for Nix. See [this fix](https://github.com/NixOS/nixpkgs/commit/c68cb4ce9bb887f349caa2a1178c3d356c5ded77) that was made just before 3.26 was released. If you update to the latest version I'm pretty sure it will work and `task --version` will output correctly (I haven't testing this myself).
